SystemManager/Src/SystemManager.cpp

```diff
@@ -0,0 +1,43 @@
+//
+// Created by Gagan Deep Singh on 2023-07-08.
+// Edited by Navtaj Hundal on 2023-07-23.
+//
+
+#include "SystemManager.hpp"
+#include "sbus_receiver.hpp"
+#include "sbus_defines.h"
+#include "rcreceiver_datatypes.h"
+#include "drivers_config.hpp"
+#include "independent_watchdog.h"
+#include "tim.h"
+
+SystemManager::SystemManager():
+    rcController_(&huart2),
+    throttleMotorChannel_(&htim2, TIM_CHANNEL_1),
+    yawMotorChannel_(&htim2, TIM_CHANNEL_2),
+    rollMotorChannel_(&htim2, TIM_CHANNEL_3),
+    pitchMotorChannel_(&htim2, TIM_CHANNEL_4),
+    watchdog_(&hiwdg)
+{}
+SystemManager::~SystemManager() {}
+
+void SystemManager::flyManually() {
+    for(;;){
+            updateRCInputs();
+            executeInputs();
+    }
+}
+
+void SystemManager::updateRCInputs() {
+    this->rcInputs_.assignValues(rcController_.GetRCControl());
+    watchdog_.refreshWatchdog();
```

Comment:
This watchdog only asserts that the function is called, not that we are still getting values from the RC. Is this the desired behaviour?

Reply:
Updated it so that SM only kicks the watchdog when new data is received